lib/diagnostic.js

```diff
@@ -5,17 +5,21 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function (previous, current) {
+  let previous = array[0];
+  array.slice().forEach(fucntion (e, i, a) {
+    callback(previous, e, i + 1, array);
+  });
+  return previous
 };
```

Comment:
Please review this 1 possible solution. Note that it utilizes a ternary operator.
```
const min = function (previous, current) {
  return previous < current ? previous : current;
};
```